itests/src/test/java/org/apache/unomi/itests/ProfileMergeIT.java

```diff
@@ -0,0 +1,110 @@
+public class ProfileMergeIT extends BaseIT {
+
+    @Inject @Filter(timeout = 600000)
+    protected EventService eventService;
+    @Inject @Filter(timeout = 600000)
+    protected RulesService rulesService;
+    @Inject @Filter(timeout = 600000)
+    protected DefinitionsService definitionsService;
+
+    private final static String TEST_EVENT_TYPE = "mergeProfileTestEventType";
+    private final static String TEST_RULE_ID = "mergeOnPropertyTest";
+    private final static String TEST_PROFILE_ID = "mergeOnPropertyTestProfileId";
+
+    @After
+    public void after() {
+        // cleanup created data
+        rulesService.removeRule(TEST_RULE_ID);
+    }
+
+    @Test
+    public void testProfileMergeOnPropertyAction_dont_forceEventProfileAsMaster() throws InterruptedException {
+        createMergeOnPropertyRule(false);
+
+        // A new profile should be created.
+        Assert.assertNotEquals(sendEvent().getProfile().getItemId(), TEST_PROFILE_ID);
+    }
+
+    @Test
+    public void testProfileMergeOnPropertyAction_forceEventProfileAsMaster() throws InterruptedException {
+        createMergeOnPropertyRule(true);
+
+        // No new profile should be created, instead the profile of the event should be used.
+        Assert.assertEquals(sendEvent().getProfile().getItemId(), TEST_PROFILE_ID);
+    }
+
+    private Event sendEvent() {
+        Profile profile = new Profile();
+        profile.setProperties(new HashMap<>());
+        profile.setItemId(TEST_PROFILE_ID);
+        profile.setProperty("j:nodename", "michel");
+        profile.getSystemProperties().put("mergeIdentifier", "jose");
+        Event testEvent = new Event(TEST_EVENT_TYPE, null, profile, null, null, profile, new Date());
+        eventService.send(testEvent);
+        return testEvent;
+    }
+
+    private void createMergeOnPropertyRule(boolean forceEventProfileAsMaster) throws InterruptedException {
+        Rule mergeOnPropertyTestRule = new Rule();
+        mergeOnPropertyTestRule.setMetadata(new Metadata(null, TEST_RULE_ID, TEST_RULE_ID, "Test rule for testing MergeProfilesOnPropertyAction"));
+
+        Condition condition = new Condition(definitionsService.getConditionType("eventTypeCondition"));
+        condition.setParameter("eventTypeId", TEST_EVENT_TYPE);
+        mergeOnPropertyTestRule.setCondition(condition);
+
+        final Action mergeProfilesOnPropertyAction = new Action( definitionsService.getActionType( "mergeProfilesOnPropertyAction"));
+        mergeProfilesOnPropertyAction.setParameter("mergeProfilePropertyValue", "eventProperty::target.properties(j:nodename)");
+        mergeProfilesOnPropertyAction.setParameter("mergeProfilePropertyName", "mergeIdentifier");
+        mergeProfilesOnPropertyAction.setParameter("forceEventProfileAsMaster", forceEventProfileAsMaster);
+        mergeOnPropertyTestRule.setActions(Collections.singletonList(mergeProfilesOnPropertyAction));
+
+        rulesService.setRule(mergeOnPropertyTestRule);
```

Comment:
For clarity, I would have use that method to build the rule, but register in by the rule service in the test by itself. (this is a really cosmetic change)